### PR TITLE
Run a custom deploy hook script defined in DEPLOY_HOOK environment variable

### PR DESCRIPTION
**Pull request by racbart**
I needed to share generated certificates with nginx, but there were no easy way to do it as nginx expects different directory/file structure. This is a popular need, as can be seen in #16.

Solution recommended in the issue, to put a script in `/etc/letsencrypt/renewal-hooks/deploy` does not help much, because scripts from this directory are executed only on certificate renewal. They are **not** executed on certificate creation.

This pull requests introduces `DEPLOY_HOOK` environment variable allowing to execute a custom deploy hook every time a certificate is created or renewed.

It's best described by a change in README (also included in the commit), so I paste it below for better readability: 

---

### Run a custom deploy hook script

You can specify a script or a command to execute after a certificate is created or renewed, by specifying `DEPLOY_HOOK` environment variable. This is useful if you want to copy certificates someplace else or need to reorganize file structure.

All standard environment variables will be available in your script, as well as two new variables set by certbot:
* `RENEWED_LINEAGE` - directory with certificate files (e.g. `/etc/letsencrypt/live/domain`)
* `RENEWED_DOMAINS` - list of domains for the certificate, separated by space 

Example: copying all new or renewed certificates to a single directory with `domain.crt` and `domain.key` filenames, making it easily usable with nginx:

Create deploy-hook.sh file and make it executable. 

```bash
#!/bin/sh
mkdir -p "/etc/nginx/certs"
cd "/etc/nginx/certs"
for domain in ${RENEWED_DOMAINS}; do
    cp "${RENEWED_LINEAGE}/fullchain.pem" "${domain}.crt"
    cp "${RENEWED_LINEAGE}/privkey.pem" "${domain}.key"
    chown $CERTS_USER_OWNER:$CERTS_GROUP_OWNER "${domain}.*"
    chmod $CERTS_FILES_MODE "${domain}.*"
done
```   

Execute:
```bash
docker run \
	--name letsencrypt-dns \
	--volume /etc/letsencrypt/domains.conf:/etc/letsencrypt/domains.conf \
	--volume /etc/letsencrypt/deploy-hook.sh:/usr/local/bin/create-nginx-certs \
	--volume /var/docker-data/letsencrypt:/etc/letsencrypt \
	--volume /var/docker-data/nginx:/etc/nginx/certs \
	--env 'LETSENCRYPT_USER_MAIL=admin@example.com' \
	--env 'LEXICON_PROVIDER=cloudflare' \
	--env 'LEXICON_CLOUDFLARE_USERNAME=my_cloudflare_email' \
	--env 'LEXICON_CLOUDFLARE_TOKEN=my_cloudflare_global_api_key' \
	--env 'DEPLOY_HOOK=create-nginx-certs' \
	adferrand/letsencrypt-dns
```

### CODE REVIEW

File: files/deploy-hook.sh [hunk 1]
```diff
@@ -14,3 +14,7 @@ fi
 find $RENEWED_LINEAGE ${RENEWED_LINEAGE/live/archive} -type d -exec chmod "$CERTS_DIRS_MODE" {} +
 find $RENEWED_LINEAGE ${RENEWED_LINEAGE/live/archive} -type f -exec chmod "$CERTS_FILES_MODE" {} +
 chown -R $CERTS_USER_OWNER:$CERTS_GROUP_OWNER $RENEWED_LINEAGE ${RENEWED_LINEAGE/live/archive}
+
+if [ ! -z "$DEPLOY_HOOK" ]; then
+    sh -c ${DEPLOY_HOOK}
```
Comment (adferrand) [inline files/deploy-hook.sh]: Calling `${DEPLOY_HOOK}` will work great for one-word commands, but not for more advanced commands (like an executable with flags). If you call `sh -c "${DEPLOY_HOOK}", you will allow to have these kind of commands.

File: files/deploy-hook.sh [hunk 2]
```diff
@@ -14,3 +14,7 @@ fi
...
+fi
```
Comment (adferrand) [inline files/deploy-hook.sh]: A new line is missing here.

**Comment by racbart** [comment]
Thanks for the comments! Fixed.

**Comment by adferrand** [comment]
Great ! I merge and prepare a release.

**Pull request merged**